src/components/Cart/BillingForm.tsx

```diff
@@ -0,0 +1,168 @@
+
+  const countriesOptions = useMemo(
+    () => transformCountriesToOptions(countries),
+    [countries],
+  );
+
+  const citiesOptions = useMemo(
+    () => transformCitiesToOptions(cities),
+    [cities],
+  );
+
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className="flex flex-col gap-7 max-w-[500px]"
+    >
+      <div className="flex flex-col gap-4">
+        <FormField
+          label="First name"
+          placeholder="firstName"
+          {...register("firstName")}
+          error={errors.firstName && errors.firstName.message}
+        />
+        <FormField
+          label="Last name"
+          placeholder="lastName"
+          {...register("lastName")}
+          error={errors.lastName && errors.lastName.message}
+        />
+        <FormField
+          label="Email"
+          placeholder="email"
+          {...register("email")}
+          error={errors.email && errors.email.message}
+        />
+        <FormField
+          label="Phone number"
+          placeholder="phoneNumber"
+          {...register("phoneNumber")}
+          error={errors.phoneNumber && errors.phoneNumber.message}
+        />
+        <FormField
+          label="Address"
+          placeholder="address"
+          {...register("address")}
+          error={errors.address && errors.address.message}
+        />
+        <Controller
+          control={control}
+          name="country"
+          render={({ field }) => {
+            return (
+              <FormControl
+                variant="standard"
+                className={twMerge("items-start")}
+              >
+                <Label>{"State / Country"}</Label>
+                <Autocomplete
+                  options={countriesOptions}
+                  value={field.value}
+                  onChange={field.onChange}
+                  inputRef={field.ref}
+                  placeholder="country"
+                />
+                {errors.country && (
+                  <span className="text-red-500">{errors.country.message}</span>
+                )}
+              </FormControl>
+            );
+          }}
+        />
+        <Controller
+          control={control}
+          name="city"
+          render={({ field }) => {
+            return (
+              <FormControl
+                variant="standard"
+                className={twMerge("items-start")}
+              >
+                <Label>{"Town / City"}</Label>
+                <Autocomplete
+                  options={citiesOptions}
+                  value={field.value}
+                  onChange={field.onChange}
+                  inputRef={field.ref}
+                  placeholder={country ? "city" : "select country"}
+                  disabled={!country}
+                />
+                {errors.city && (
+                  <span className="text-red-500">{errors.city.message}</span>
+                )}
+              </FormControl>
+            );
+          }}
+        />
```

Comment:
Put autocomplete into separate reusable component please. And try to use the same approach everywhere, via `Controller` or `register`

Reply:
Autocomplete is already separated in the next task. Controller approach is done in this task.